src/Simple.OData.Client.V4.Adapter/CommandFormatter.cs

```diff
@@ -127,84 +152,68 @@ protected override void FormatInlineCount(IList<string> commandClauses)
             return text;
         }
 
-        private IList<string> SelectPathSegmentColumns(
-            IList<string> columns, string path, EntityCollection collection, IList<string> excludePaths = null)
+        private static ODataExpandAssociation MergeOrderByColumns(ODataExpandAssociation expandAssociation, KeyValuePair<string, bool> orderByColumn)
         {
-            if (string.IsNullOrEmpty(path))
+            if (string.IsNullOrEmpty(orderByColumn.Key))
+                return expandAssociation;
+            
+            var segments = orderByColumn.Key.Split('/');
+            if (segments[0] != expandAssociation.Name)
+                return expandAssociation;
+
+            var result = expandAssociation.Clone();
+            MergeOrderByColumns(result, segments.Skip(1).ToArray(), orderByColumn.Value);
```

Comment:
This should be skipped if segments.Length == 1, avoiding the LINQ projection and ToArray call.

Reply:
I have replaced the calls of .Skip(1).ToArray() with iteration over the segments